bin/p2-setup-runtime/main.go

```diff
@@ -0,0 +1,191 @@
+	tlsCert       = app.Flag("tls-cert", "File containing tls cert").ExistingFile()
+	tlsKey        = app.Flag("tls-key", "File containing tls key").ExistingFile()
+)
+
+func dockerClient() (*dockerclient.Client, error) {
+	var client *dockerclient.Client
+	var err error
+
+	if *tlsCA != "" && *tlsCert != "" && *tlsKey != "" {
+		client, err = dockerclient.NewClientWithOpts(
+			dockerclient.WithTLSClientConfig(*tlsCA, *tlsCert, *tlsKey),
+			dockerclient.WithHost(*dockerUrl),
+			dockerclient.WithVersion(*dockerVersion),
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client, err = dockerclient.NewClientWithOpts(
+			dockerclient.WithHost(*dockerUrl),
+			dockerclient.WithVersion(*dockerVersion),
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return client, err
+}
+
+func initPod(manifest manifest.Manifest) (*pods.Pod, error) {
+	nodeName, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	fetcher := uri.DefaultFetcher
+	podReadOnlyPolicy := pods.NewReadOnlyPolicy(false, nil, nil)
+	podFactory := pods.NewFactory(*podRoot, types.NodeName(nodeName), fetcher, *requireFile, podReadOnlyPolicy)
+
+	dockerClient, err := dockerClient()
+	if err != nil {
+		return nil, err
+	}
+
+	podFactory.SetDockerClient(*dockerClient)
+	pod := podFactory.NewLegacyPod(manifest.ID())
+	return pod, nil
+}
+
+func main() {
+	app.Version(version.VERSION)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	pods.Log.Logger.Formatter = &logrus.TextFormatter{
+		DisableTimestamp: false,
+		FullTimestamp:    true,
+		TimestampFormat:  "23:59:59.999",
+	}
+	logger := pods.Log
+
+	// read manifest
+	manifest, err := manifest.FromURI(*manifestURI)
+	if err != nil {
+		logger.WithError(err).WithField("manifest", *manifestURI).Fatalln("Unable to parse manifest file")
+	}
+
+	// list of hook names for which failure is fatal
+	hooksRequired := []string{}
+
+	// setup hooks context
+	hookLogger := logging.DefaultLogger
+	hookLogger.Logger.SetLevel(logrus.ErrorLevel)
+	auditLogger := hooks.NewFileAuditLogger(&hookLogger)
+	hookContext := hooks.NewContext(*hooksDir, *podRoot, &hookLogger, auditLogger)
+
+	logger = logger.WithField("app", manifest.ID())
+
+	// initialize pod
+	pod, err := initPod(manifest)
+	if err != nil {
+		logger.WithError(err).Fatalln("Failure initializing pod environment")
+
+	}
+
+	launchables, err := pod.Launchables(manifest)
+	if err != nil {
+		logger.WithError(err).Fatalln("Failure getting launchables from manifest")
+	}
+
+	if len(launchables) == 0 {
+		logger.Fatalln("There are no launchables in manifest")
+	}
+
+	//run hooks BeforeInstall
+	logger.Infoln("Running before_install hooks")
+	err = hookContext.RunHookType(hooks.BeforeInstall, pod, manifest, hooksRequired)
+	if err != nil {
+		logger.WithError(err).Fatalln("Failure running before_install hook(s)")
+	}
+
+	// run hooks AfterInstall
+	logger.Infoln("Running after_install hooks")
+	err = hookContext.RunHookType(hooks.AfterInstall, pod, manifest, hooksRequired)
```

Comment:
You're able to just run the before and after install hooks because we assume the install itself has already been run by `p2-launch`, right?

Reply:
yes, we assume pod is already installed. so we run all hooks as needed